packages/backend-core/src/plugin/utils.ts

```diff
@@ -82,6 +90,55 @@ function validateDatasource(schema: any) {
   runJoi(validator, schema)
 }
 
+function validateAutomation(schema: any) {
```

Comment:
These validations are quite heavy. Do we usually write unit tests for it?

Reply:
We haven't in the past, but could easily add something in to run through the datasource/automation validations with our internal ones, the internal ones should all pass so would be an easy test case to write.